plugins/flytekit-duckdb/flytekitplugins/duckdb/task.py

```diff
@@ -32,10 +33,17 @@ def __init__(
             name: Name of the task
             query: DuckDB query to execute
             inputs: The query parameters to be used while executing the query
+            md_token_name: Name of the secret containing a MotherDuck authentication token
         """
+        if md_token_name is None:
+            # create an in-memory database that's non-persistent
+            self._con = duckdb.connect(":memory:")
+        else:
+            # connect to motherduck
+            md_token = current_context().secrets.get(md_token_name)
```

Comment:
i don't think this will work for k8s secrets. we should add support for group, key, and group version.

Reply:
ah thanks! Let me know if the latest commit is what you're looking for